scripts/check-repo-names.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+
+# Copyright (c) 2024 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+set -euo pipefail
+
+SCRIPTNAME=${0##*/}
+
+rename_script="scripts/check-repo-names.sh"
+
+function check_patterns() {
```

Comment:
can we make this run against the splice code dump directly? This script can call `scripts/copy-to-splice.sh /tmp/splice-dump` or something and cd there

Reply:
Sure, you can pass `-s path/to/sourcerepo` and it will do that as of 17f92a6f6f4f11d0da369ae8556ce81019a31e28.